server/src/main/java/org/opensearch/index/store/remote/metadata/RemoteSegmentMetadata.java

```diff
@@ -38,16 +38,19 @@ public class RemoteSegmentMetadata {
 
     private final byte[] segmentInfosBytes;
 
+    private final long primaryTerm;
```

Comment:
Why do we need this?

Reply:
replied [here](https://github.com/opensearch-project/OpenSearch/pull/6788/files#r1242903727).